### ISSUE TITLE
composer install failure

### ISSUE THREAD

**Issue by daniels123kid** [aimeos/aimeos#1]
please am sorry to be the one with an issue here may be am not getting it right but on the documentaion you listed commands that i have run and am getting an error and dont see how to get over it. here is the error:----

$ composer create-project aimeos/aimeos myshoppingsite
Installing aimeos/aimeos (2017.10.1)
  - Installing aimeos/aimeos (2017.10.1): Loading from cache
Created project in myshoppingsite
> @php -r "file_exists('.env') || copy('.env.example', '.env');"
> mkdir public/files public/preview public/uploads
The syntax of the command is incorrect.
Script mkdir public/files public/preview public/uploads handling the post-root-package-install event returned with error code 1

checked all the prerequisites and i have them
thanks i would love to check this app and integrate it on my site

**Comment by aimeos** [comment]
Are you developing in a Windows environment?

**Comment by aimeos** [comment]
We've changed the `mkdir` command so it should now in Windows environments too. You can test with:
`composer create-project -d dev aimeos/aimeos myshoppingsite`

**Comment by daniels123kid** [comment]
thank you sir for your time and response.
yes am using windows environment for my development with xampp server.
am stilling getting the error when i use the new:
 composer create-project -d dev aimeos/aimeos myshoppingsite
"
DevHacker@DevHacker-PC MINGW64 /c/xampp/htdocs/ant-vel
$ composer create-project -d dev aimeos/aimeos myshoppingsite

  [RuntimeException]
  Invalid working directory specified, dev does not exist.
"

**Comment by aimeos** [comment]
Sorry, my fault! It's

`composer create-project -s dev aimeos/aimeos myshoppingsite`

for the development branch.

**Comment by daniels123kid** [comment]
Thanks you again i cant appreciate you enough for your time , 
it installed 
but having issue with defaults of password yet in my setup doesn't have password .
thank you very much

**Comment by aimeos** [comment]
What does that mean? At the end of the setup process you are asked to enter an e-mail address and a password for the administration interface

**Comment by daniels123kid** [comment]
at the end asks for database credentials like in the .env so the problem is
am not using any passwords for phpmyadmin.
the installation will ask for database name and username and when it comes to password i want to live it blank because i see by default it has secret in brackets so i am supposed to press enter key for defaults yet i also want to mean no password the installations fails saying (user access denied password set:Yes)

i dnot see where i can opt for no password option since by default its secret in brackets.

**Comment by aimeos** [comment]
A database password is required. Create a new database user with a password, then it will work.

**Comment by AhmedHassanEldesoky** [comment]
i don't know what's the reason for that error ?
![errr](https://user-images.githubusercontent.com/32817087/31585819-82f88942-b1c8-11e7-9a57-89e75ac8cd1e.PNG)

**Comment by aimeos** [comment]
@AhmedHassanEldesoky We improved the output when a command fails. Can you please test and run
`composer create-project -s dev aimeos/aimeos`

**Comment by daniels123kid** [comment]
i have setup a new user with default credentials but i still get an error

Authentication scaffolding generated successfully.
> App\Composer::configure
Database setup
- DB_CONNECTION (mysql): mysql
- DB_HOST (127.0.0.1):
- DB_PORT (3306):
- DB_DATABASE (homestead): idukai
- DB_USERNAME (homestead): dan
- DB_PASSWORD (secret): secret
Mail setup
- MAIL_DRIVER (smtp):
- MAIL_HOST (smtp.mailtrap.io):
- MAIL_PORT (2525):
- MAIL_USERNAME ():
- MAIL_PASSWORD ():
- MAIL_ENCRYPTION ():
> App\Composer::setup
PHP Fatal error:  Uncaught TypeError: Argument 6 passed to Symfony\Component\Process\Process::__construct() must be of the type array, null given, called in C:\xampp\htdocs\eduka\vendor\symfony\process\ProcessBuilder.php on line 270 and defined in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/process/Process.php:141
Stack trace:
#0 C:\xampp\htdocs\eduka\vendor\symfony\process\ProcessBuilder.php(270): Symfony\Component\Process\Process->__construct(Array, NULL, Array, NULL, 60, NULL)
#1 C:\xampp\htdocs\eduka\app\Composer.php(144): Symfony\Component\Process\ProcessBuilder->getProcess()
#2 C:\xampp\htdocs\eduka\app\Composer.php(112): App\Composer::executeCommand(Object(Composer\Script\Event), 'aimeos:setup', Array)
#3 phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php(282): App\Composer::setup(Object(Composer\Script\Event))
#4 phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php(205): Composer\EventDispatche in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/process/Process.php on line 141

Fatal error: Uncaught TypeError: Argument 6 passed to Symfony\Component\Process\Process::__construct() must be of the type array, null given, called in C:\xampp\htdocs\eduka\vendor\symfony\process\ProcessBuilder.php on line 270 and defined in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/process/Process.php:141
Stack trace:
#0 C:\xampp\htdocs\eduka\vendor\symfony\process\ProcessBuilder.php(270): Symfony\Component\Process\Process->__construct(Array, NULL, Array, NULL, 60, NULL)
#1 C:\xampp\htdocs\eduka\app\Composer.php(144): Symfony\Component\Process\ProcessBuilder->getProcess()
#2 C:\xampp\htdocs\eduka\app\Composer.php(112): App\Composer::executeCommand(Object(Composer\Script\Event), 'aimeos:setup', Array)
#3 phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php(282): App\Composer::setup(Object(Composer\Script\Event))
#4 phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/EventDispatcher/EventDispatcher.php(205): Composer\EventDispatche in phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/process/Process.php on line 141

**Comment by daniels123kid** [comment]
![error](https://user-images.githubusercontent.com/24682769/31589482-d4307152-b20a-11e7-9264-c808f59611d0.PNG)
 thats the error after using the required credentials

**Comment by aimeos** [comment]
Strange, seems like the Symfony ProcessBuilder class is buggy on Windows platforms. I've added a workaround and hope it helps. Unfortunately, nobody of us uses Windows for development and we can't test it ourselves, so can you please run composer again?

**Comment by daniels123kid** [comment]
still the error is the same 
if i may ask what dev env't are you developing from mac of linux

**Comment by aimeos** [comment]
I've seen that the Symfony ProcessBuilder class is deprecated and will be removed. Thus, I've rewritten that part. Can you execute `composer create-project -s dev aimeos/aimeos` again please?

Most of us use Linux, some Mac but it should work on Windows too. If you test it, we will fix it until it finally works for you :-)

**Comment by daniels123kid** [comment]
thank you very much sir

its working in windows perfectly

just asking for your opinion here between linux and windows for purposes of development which one is best or better. i have no doubt using linux but i only had small confusion setting up homestead and vagrant the reason was i wasnt familiar with ssh and how the magic happens setting up yml stuff but with your opinion i think i would cling on it. thanks gain it worked this time round.

**Comment by aimeos** [comment]
Most developers prefer a Linux like environment because 99% of all web hosting is Linux based. That prevents some problems when developing locally on Windows and then it doesn't work on the live server.

**Comment by AhmedHassanEldesoky** [comment]
Unfortunately, still not working with me , everything was better until the installation reaches to this point "Rebuilding index for demo data" , i have been trying more than one time and every time it gives me the same error.
![err](https://user-images.githubusercontent.com/32817087/31605362-5822e174-b266-11e7-81ed-0420207bb37e.PNG)

**Comment by aimeos** [comment]
There's no error output so maybe the process terminated with a segmentation fault. Which PHP version do you use?

**Comment by AhmedHassanEldesoky** [comment]
PHP 7.1.9

**Comment by aimeos** [comment]
Can you please execute this command directly on the command line:
`php artisan aimeos:setup --option=setup/default/demo:1`

If that fails too, we would need the output. You can also try without demo data:
`php artisan aimeos:setup`

**Comment by daniels123kid** [comment]
hey hope your doing well
am building my project on top of the aimeos laravel shop application so am having challanges migrating the newly added tables on my database migrations. i decided to use a new database with new credentials after the installation using composer create-project -s dev aimeos/aimeos

there are irregularities with the folder structure where the migration that i originally migrated during the installation are not in the migrations folder. where the pages are routed not from the view/resources folder but in the public folder of laravel.

i have added new migrations so i try running php artisan migrate only the newly added tables were added. 
how do i get to migrate both the original tables created during the installation and the new ones i just custom created

**Comment by aimeos** [comment]
If you have questions about Aimeos, please use our forum: https://aimeos.org/help/laravel-package-f18/

**Comment by daniels123kid** [comment]
![error after](https://user-images.githubusercontent.com/24682769/31658877-00ea5ea2-b33b-11e7-998a-d875c78968c0.PNG)

**Comment by AhmedHassanEldesoky** [comment]
it's working now :+1:
thanks for your time sir

**Comment by aimeos** [comment]
@AhmedHassanEldesoky Including or excluding the demo data?

**Comment by daniels123kid** [comment]
excluding demo data

**Comment by AhmedHassanEldesoky** [comment]
excluding the demo data

**Comment by aimeos** [comment]
@daniels123kid @AhmedHassanEldesoky Is there any error message at the end if execution including demo data fails?

**Comment by daniels123kid** [comment]
![new error](https://user-images.githubusercontent.com/24682769/31679549-2025072e-b37a-11e7-9c16-d9fd6dca5d07.PNG)
 that error is after i added my custom migrations and i run php artisan aimeos setup command
so i wanted to change some of the original migrations of the user table that comes with aimeos package i cant see where to get it from

**Comment by aimeos** [comment]
Seems like you have a conflicting declaration of the users.id column.

But that's not the problem that the setup process terminated at the catalog rebuild stage. We are interested if any error output is after the catalog rebuild to solve that problem

**Comment by daniels123kid** [comment]
ok sir but where are the original migration files of aimeos?

**Comment by aimeos** [comment]
Aimeos also uses DBAL for schema migration but not Laravel style:
https://github.com/aimeos/aimeos-core/tree/master/lib/mshoplib/setup/default/schema

**Comment by daniels123kid** [comment]
thanks

**Comment by ranaumer8344** [comment]
> Sorry, my fault! It's
> 
> `composer create-project -s dev aimeos/aimeos myshoppingsite`
> 
> for the development branch.

it's not working:

Installing aimeos/aimeos (dev-master 13429645545e77e5d44e504a3924b12dad86bd67)
  - Installing aimeos/aimeos (dev-master 1342964): Cloning 1342964554
Cloning failed using an ssh key for authentication, enter your GitHub credentials to access private repos
Head to https://github.com/settings/tokens/new?scopes=repo&description=Composer+on+RanaUmer+2020-02-26+1345
to retrieve a token. It will be stored in "C:/Users/PAKISTAN/AppData/Roaming/Composer/auth.json" for future use by Composer.
Token (hidden):

**Comment by aimeos** [comment]
Don't use the dev-master version if you don't need it for testing new feature. Instead, use
`composer create-project aimeos/aimeos myshop`

**Comment by ranaumer8344** [comment]
Thanks for replying! 
While installing Aimeos I am getting this error after connecting database in cl
Script Php artisan vendor: publish handling the post-update-cmd event returned with error code 1

**Comment by aimeos** [comment]
Which MySQL version do you use?
Are there more details about the problem available?

**Comment by ranaumer8344** [comment]
I am using MySQL version: 5.7.28

here is the complete detail of installation:

PS C:\wamp64\www> composer create-project aimeos/aimeos shop
Installing aimeos/aimeos (2019.10.2)
  - Installing aimeos/aimeos (2019.10.2): Loading from cache
Created project in shop
> @php -r "file_exists('.env') || copy('.env.example', '.env');"
> @php -r "mkdir('public/files'); mkdir('public/preview'); mkdir('public/uploads');"
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 104 installs, 0 updates, 0 removals
  - Installing composer/installers (v1.8.0): Loading from cache
  - Installing symfony/polyfill-ctype (v1.14.0): Loading from cache
  - Installing phpoption/phpoption (1.7.2): Loading from cache
  - Installing vlucas/phpdotenv (v3.6.0): Loading from cache
  - Installing symfony/css-selector (v5.0.4): Loading from cache
  - Installing tijsverkoyen/css-to-inline-styles (2.2.2): Loading from cache
  - Installing symfony/polyfill-php72 (v1.14.0): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.14.0): Loading from cache
  - Installing symfony/var-dumper (v4.4.4): Loading from cache
  - Installing symfony/routing (v4.4.4): Loading from cache
  - Installing symfony/process (v4.4.4): Loading from cache
  - Installing psr/log (1.1.2): Loading from cache
  - Installing symfony/polyfill-php73 (v1.14.0): Loading from cache
  - Installing symfony/polyfill-intl-idn (v1.14.0): Loading from cache
  - Installing symfony/mime (v5.0.4): Loading from cache
  - Installing symfony/http-foundation (v4.4.4): Loading from cache
  - Installing symfony/event-dispatcher-contracts (v1.1.7): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing symfony/event-dispatcher (v4.4.4): Loading from cache
  - Installing symfony/debug (v4.4.4): Loading from cache
  - Installing symfony/error-handler (v4.4.4): Loading from cache
  - Installing symfony/http-kernel (v4.4.4): Loading from cache
  - Installing symfony/finder (v4.4.4): Loading from cache
  - Installing symfony/service-contracts (v2.0.1): Loading from cache
  - Installing symfony/console (v4.4.4): Loading from cache
  - Installing symfony/polyfill-iconv (v1.14.0): Loading from cache
  - Installing doctrine/lexer (1.2.0): Loading from cache
  - Installing egulias/email-validator (2.1.17): Loading from cache
  - Installing swiftmailer/swiftmailer (v6.2.3): Loading from cache
  - Installing paragonie/random_compat (v9.99.99): Loading from cache
  - Installing ramsey/uuid (3.9.3): Loading from cache
  - Installing psr/simple-cache (1.0.1): Loading from cache
  - Installing opis/closure (3.5.1): Loading from cache
  - Installing symfony/translation-contracts (v2.0.1): Loading from cache
  - Installing symfony/translation (v4.4.4): Loading from cache
  - Installing nesbot/carbon (2.30.0): Loading from cache
  - Installing monolog/monolog (2.0.2): Loading from cache
  - Installing league/flysystem (1.0.64): Loading from cache
  - Installing dragonmantank/cron-expression (v2.3.0): Loading from cache
  - Installing doctrine/inflector (1.3.1): Loading from cache
  - Installing league/commonmark (1.3.0): Loading from cache
  - Installing laravel/framework (v6.17.1): Loading from cache
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing zendframework/zend-diactoros (1.8.7): Loading from cache
  - Installing jakub-onderka/php-console-color (v0.2): Loading from cache
  - Installing nikic/php-parser (v4.3.0): Loading from cache
  - Installing fideloper/proxy (4.3.0): Loading from cache
  - Installing jakub-onderka/php-console-highlighter (v0.4): Loading from cache
  - Installing dnoegel/php-xdg-base-dir (v0.1.1): Loading from cache
  - Installing psy/psysh (v0.9.12): Loading from cache
  - Installing laravel/tinker (v1.0.10): Loading from cache
  - Installing laravel/ui (v1.2.0): Loading from cache
  - Installing doctrine/event-manager (1.1.0): Loading from cache
  - Installing doctrine/cache (1.10.0): Loading from cache
  - Installing doctrine/dbal (v2.10.1): Loading from cache
  - Installing aimeos/aimeos-core (2019.10.18): Loading from cache
  - Installing aimeos/ai-controller-frontend (2019.10.9): Loading from cache
  - Installing aimeos/ai-controller-jobs (2019.10.10): Loading from cache
  - Installing aimeos/ai-client-jsonapi (2019.10.4): Loading from cache
  - Installing aimeos/ai-client-html (2019.10.13): Loading from cache
  - Installing aimeos/ai-admin-jsonadm (2019.10.4): Loading from cache
  - Installing aimeos/ai-admin-jqadm (2019.10.11): Loading from cache
  - Installing aimeos/ai-laravel (2019.10.5): Loading from cache
  - Installing aimeos/ai-gettext (2019.10.1): Loading from cache
  - Installing aimeos/ai-swiftmailer (2019.10.1): Loading from cache
  - Installing symfony/psr-http-message-bridge (v1.3.0): Loading from cache
  - Installing aimeos/aimeos-laravel (2019.10.2): Loading from cache
  - Installing scrivo/highlight.php (v9.18.1.0): Loading from cache
  - Installing filp/whoops (2.7.1): Loading from cache
  - Installing facade/ignition-contracts (1.0.0): Loading from cache
  - Installing facade/flare-client-php (1.3.1): Loading from cache
  - Installing facade/ignition (1.16.0): Loading from cache
  - Installing fzaninotto/faker (v1.9.1): Loading from cache
  - Installing hamcrest/hamcrest-php (v2.0.0): Loading from cache
  - Installing mockery/mockery (1.3.1): Loading from cache
  - Installing nunomaduro/collision (v3.0.1): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/type (1.1.3): Loading from cache
  - Installing sebastian/resource-operations (2.0.1): Loading from cache
  - Installing sebastian/recursion-context (3.0.0): Loading from cache
  - Installing sebastian/object-reflector (1.1.1): Loading from cache
  - Installing sebastian/object-enumerator (3.0.3): Loading from cache
  - Installing sebastian/global-state (3.0.0): Loading from cache
  - Installing sebastian/exporter (3.1.2): Loading from cache
  - Installing sebastian/environment (4.2.3): Loading from cache
  - Installing sebastian/diff (3.0.2): Loading from cache
  - Installing sebastian/comparator (3.0.2): Loading from cache
  - Installing phpunit/php-timer (2.1.2): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/php-file-iterator (2.0.2): Loading from cache
  - Installing theseer/tokenizer (1.1.3): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-token-stream (3.1.1): Loading from cache
  - Installing phpunit/php-code-coverage (7.0.10): Loading from cache
  - Installing doctrine/instantiator (1.3.0): Loading from cache
  - Installing phpdocumentor/reflection-common (2.0.0): Loading from cache
  - Installing phpdocumentor/type-resolver (1.0.1): Loading from cache
  - Installing webmozart/assert (1.7.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (5.1.0): Loading from cache
  - Installing phpspec/prophecy (v1.10.2): Loading from cache
  - Installing phar-io/version (2.0.1): Loading from cache
  - Installing phar-io/manifest (1.0.3): Loading from cache
  - Installing myclabs/deep-copy (1.9.5): Loading from cache
  - Installing phpunit/phpunit (8.5.2): Loading from cache
symfony/routing suggests installing symfony/config (For using the all-in-one router or any loader)
symfony/routing suggests installing symfony/yaml (For using the YAML loader)
symfony/routing suggests installing symfony/expression-language (For using expression matching)
symfony/routing suggests installing doctrine/annotations (For using the annotation loader)
symfony/event-dispatcher-contracts suggests installing psr/event-dispatcher
symfony/event-dispatcher suggests installing symfony/dependency-injection
symfony/http-kernel suggests installing symfony/browser-kit
symfony/http-kernel suggests installing symfony/config
symfony/http-kernel suggests installing symfony/dependency-injection
symfony/service-contracts suggests installing symfony/service-implementation
symfony/console suggests installing symfony/lock
swiftmailer/swiftmailer suggests installing true/punycode (Needed to support internationalized email addresses, if ext-intl is not installed)
paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
ramsey/uuid suggests installing ext-libsodium (Provides the PECL libsodium extension for use with the SodiumRandomGenerator)
ramsey/uuid suggests installing ext-uuid (Provides the PECL UUID extension for use with the PeclUuidTimeGenerator and PeclUuidRandomGenerator)
ramsey/uuid suggests installing moontoast/math (Provides support for converting UUID to 128-bit integer (in string form).)
ramsey/uuid suggests installing ramsey/uuid-console (A console application for generating UUIDs with ramsey/uuid)
ramsey/uuid suggests installing ramsey/uuid-doctrine (Allows the use of Ramsey\Uuid\Uuid as Doctrine field type.)
ramsey/uuid suggests installing paragonie/random-lib (Provides RandomLib for use with the RandomLibAdapter)
symfony/translation suggests installing symfony/config
symfony/translation suggests installing symfony/yaml
monolog/monolog suggests installing graylog2/gelf-php (Allow sending log messages to a GrayLog2 server)
monolog/monolog suggests installing doctrine/couchdb (Allow sending log messages to a CouchDB server)
monolog/monolog suggests installing ruflin/elastica (Allow sending log messages to an Elastic Search server)
monolog/monolog suggests installing elasticsearch/elasticsearch (Allow sending log messages to an Elasticsearch server via official client)
monolog/monolog suggests installing php-amqplib/php-amqplib (Allow sending log messages to an AMQP server using php-amqplib)
monolog/monolog suggests installing ext-amqp (Allow sending log messages to an AMQP server (1.0+ required))
monolog/monolog suggests installing ext-mongodb (Allow sending log messages to a MongoDB server (via driver))
monolog/monolog suggests installing mongodb/mongodb (Allow sending log messages to a MongoDB server (via library))
monolog/monolog suggests installing aws/aws-sdk-php (Allow sending log messages to AWS services like DynamoDB)
monolog/monolog suggests installing rollbar/rollbar (Allow sending log messages to Rollbar)
monolog/monolog suggests installing php-console/php-console (Allow sending log messages to Google Chrome)
league/flysystem suggests installing league/flysystem-eventable-filesystem (Allows you to use EventableFilesystem)
league/flysystem suggests installing league/flysystem-rackspace (Allows you to use Rackspace Cloud Files)
league/flysystem suggests installing league/flysystem-azure (Allows you to use Windows Azure Blob storage)
league/flysystem suggests installing league/flysystem-webdav (Allows you to use WebDAV storage)
league/flysystem suggests installing league/flysystem-aws-s3-v2 (Allows you to use S3 storage with AWS SDK v2)
league/flysystem suggests installing league/flysystem-aws-s3-v3 (Allows you to use S3 storage with AWS SDK v3)
league/flysystem suggests installing spatie/flysystem-dropbox (Allows you to use Dropbox storage)
league/flysystem suggests installing srmklive/flysystem-dropbox-v2 (Allows you to use Dropbox storage for PHP 5 applications)
league/flysystem suggests installing league/flysystem-cached-adapter (Flysystem adapter decorator for metadata caching)
league/flysystem suggests installing ext-ftp (Allows you to use FTP server storage)
league/flysystem suggests installing league/flysystem-sftp (Allows you to use SFTP server storage via phpseclib)
league/flysystem suggests installing league/flysystem-ziparchive (Allows you to use ZipArchive adapter)
laravel/framework suggests installing ext-memcached (Required to use the memcache cache driver.)
laravel/framework suggests installing ext-pcntl (Required to use all features of the queue worker.)
laravel/framework suggests installing ext-posix (Required to use all features of the queue worker.)
laravel/framework suggests installing ext-redis (Required to use the Redis cache and queue drivers.)
laravel/framework suggests installing aws/aws-sdk-php (Required to use the SQS queue driver, DynamoDb failed job storage and SES mail driver (^3.0).)
laravel/framework suggests installing guzzlehttp/guzzle (Required to use the Mailgun mail driver and the ping methods on schedules (^6.0|^7.0).)
laravel/framework suggests installing league/flysystem-aws-s3-v3 (Required to use the Flysystem S3 driver (^1.0).)
laravel/framework suggests installing league/flysystem-cached-adapter (Required to use the Flysystem cache (^1.0).)
laravel/framework suggests installing league/flysystem-sftp (Required to use the Flysystem SFTP driver (^1.0).)
laravel/framework suggests installing moontoast/math (Required to use ordered UUIDs (^1.1).)
laravel/framework suggests installing nyholm/psr7 (Required to use PSR-7 bridging features (^1.2).)
laravel/framework suggests installing pda/pheanstalk (Required to use the beanstalk queue driver (^4.0).)
laravel/framework suggests installing pusher/pusher-php-server (Required to use the Pusher broadcast driver (^4.0).)
laravel/framework suggests installing symfony/cache (Required to PSR-6 cache bridge (^4.3.4).)
laravel/framework suggests installing wildbit/swiftmailer-postmark (Required to use Postmark mail driver (^3.0).)
psy/psysh suggests installing ext-pcntl (Enabling the PCNTL extension makes PsySH a lot happier :))
psy/psysh suggests installing ext-posix (If you have PCNTL, you'll want the POSIX extension as well.)
psy/psysh suggests installing ext-pdo-sqlite (The doc command requires SQLite to work.)
psy/psysh suggests installing hoa/console (A pure PHP readline implementation. You'll want this if your PHP install doesn't already support readline or libedit.)      doctrine/cache suggests installing alcaeus/mongo-php-adapter (Required to use legacy MongoDB driver)
aimeos/aimeos-core suggests installing apigen/apigen (Required for generating the API documentation)
symfony/psr-http-message-bridge suggests installing nyholm/psr7 (For a super lightweight PSR-7/17 implementation)
filp/whoops suggests installing whoops/soap (Formats errors as SOAP responses)
facade/ignition suggests installing laravel/telescope (^2.0)
sebastian/global-state suggests installing ext-uopz (*)
sebastian/environment suggests installing ext-posix (*)
phpunit/php-code-coverage suggests installing ext-xdebug (^2.7.2)
phpunit/phpunit suggests installing phpunit/php-invoker (^2.0.0)
phpunit/phpunit suggests installing ext-xdebug (*)
Package zendframework/zend-diactoros is abandoned, you should avoid using it. Use laminas/laminas-diactoros instead.
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: aimeos/aimeos-laravel
Discovered Package: facade/ignition
Discovered Package: fideloper/proxy
Discovered Package: laravel/tinker
Discovered Package: laravel/ui
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Package manifest generated successfully.
> @php artisan vendor:publish --tag=public --force
Copied Directory [\vendor\aimeos\aimeos-laravel\public] To [\public\packages\aimeos\shop]
Publishing complete.
> @php artisan key:generate --ansi
Application key set successfully.
> App\Composer::configure
Database setup
- DB_CONNECTION (mysql):
- DB_HOST (127.0.0.1):
- DB_PORT (3306):
- DB_DATABASE (laravel): shop
- DB_USERNAME (root):
- DB_PASSWORD:
Mail setup
- MAIL_DRIVER (smtp):
- MAIL_HOST (smtp.mailtrap.io):
- MAIL_PORT (2525):
- MAIL_USERNAME ():
- MAIL_ENCRYPTION ():
- MAIL_PASSWORD:
> @php artisan migrate
Migration table created successfully.
Migrating: 2014_10_12_000000_create_users_table

   Illuminate\Database\QueryException  : SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes (SQL: alter table `users` add unique `users_email_unique`(`email`))

  at C:\wamp64\www\shop\vendor\laravel\framework\src\Illuminate\Database\Connection.php:669
    665|         // If an exception occurs when attempting to run a query, we'll format the error
    666|         // message to include the bindings with SQL, which will make this exception a
    667|         // lot more helpful to the developer instead of just the database's errors.
    668|         catch (Exception $e) {
  > 669|             throw new QueryException(
    670|                 $query, $this->prepareBindings($bindings), $e
    671|             );
    672|         }
    673|

  Exception trace:

  1   Doctrine\DBAL\Driver\PDOException::("SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes")
      C:\wamp64\www\shop\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOStatement.php:123

  2   PDOException::("SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 1000 bytes")
      C:\wamp64\www\shop\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOStatement.php:121

  Please use the argument -v to see more details.
Script @php artisan migrate handling the post-create-project-cmd event returned with error code 1
PS C:\wamp64\www>

**Comment by aimeos** [comment]
This explains the problem: https://github.com/laravel/framework/issues/24711#issuecomment-552141282

Please make sure the default table engine is InnoDB in your MySQL confiugration

**Comment by brigo77** [comment]
Hi,
Can someone help me with this:

when I try to run "php artisan aimeos:setup" I get this message

Aimeos\MW\Setup\Exception  : Database schema class "\Aimeos\MW\Setup\DBSchema\System" not found

[aimeos-error.txt](https://github.com/aimeos/aimeos/files/4515884/aimeos-error.txt)
I'm using laravel 5.8 and hyn/multi-tenant.

Thank you

**Comment by rahulsarafrs** [comment]
getting this error

![Uploading image.png…]()

**Comment by rahulsarafrs** [comment]
Database setup
- DB_CONNECTION (mysql):
- DB_HOST (127.0.0.1):
- DB_PORT (3306): 3308
- DB_DATABASE (laravel): alumnor
- DB_USERNAME (root):
- DB_PASSWORD: Script App\Composer::configure handling the post-create-project-cmd event terminated with an exception


  [ErrorException]
  copy(phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Helper/../Resources/bin/hiddeninp
  ut.exe): failed to open stream: phar error: "vendor/symfony/console/Resources/bin/hiddeninput.exe" is not a file in
   phar "C:/ProgramData/ComposerSetup/bin/composer.phar"


create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [<package>] [<directory>] [<version>]

**Comment by aimeos** [comment]
Seem like a composer problem. Can you download composer here and make sure this package is used:
https://getcomposer.org/download/

**Comment by rahulsarafrs** [comment]
Getting this error message on dashboard after admin login, tried enabling opcache in phpini still getting same error


PHP setting "opcache.enable" is disabled

**Comment by rahulsarafrs** [comment]
![image](https://user-images.githubusercontent.com/73284642/98095603-cd289b80-1eb0-11eb-9c20-2d9228c6a63c.png)

**Comment by aimeos** [comment]
It's only a warning. You can test Aimeos and customize it. On your production server you should enable the setting for performance reasons.

**Comment by nemesreviz** [comment]
> 
> 
> Database setup
> 
>     * DB_CONNECTION (mysql):
> 
>     * DB_HOST (127.0.0.1):
> 
>     * DB_PORT (3306): 3308
> 
>     * DB_DATABASE (laravel): alumnor
> 
>     * DB_USERNAME (root):
> 
>     * DB_PASSWORD: Script App\Composer::configure handling the post-create-project-cmd event terminated with an exception
>       [ErrorException]
>       copy(phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Helper/../Resources/bin/hiddeninp
>       ut.exe): failed to open stream: phar error: "vendor/symfony/console/Resources/bin/hiddeninput.exe" is not a file in
>       phar "C:/ProgramData/ComposerSetup/bin/composer.phar"
> 
> 
> create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--add-repository] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--remove-vcs] [--no-install] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--ask] [--] [] [] []

Having just the same error message still after updating Composer to v2.0.6 from v2.0.4.
Tried to bypass it running __artisan migrate__, but mshop_locale_site table (and maybe others also) is missing.
Any clues how to manage this? Thanks!

**Comment by aimeos** [comment]
Seems like something you should open an issue in the composer repo ...

To finish Aimeos installation without composer:
1.) Add your DB credentials to your .env file in the application directory
2.) Run those commands:
```
php artisan migrate
php artisan vendor:publish --tag=public --force
php artisan aimeos:setup --option=setup/default/demo:1
php artisan aimeos:account --super <email address>
```

**Comment by nemesreviz** [comment]
> 
> 
> Seems like something you should open an issue in the composer repo ...
> 
> To finish Aimeos installation without composer:
> 1.) Add your DB credentials to your .env file in the application directory
> 2.) Run those commands:
> 
> ```
> php artisan migrate
> php artisan vendor:publish --tag=public --force
> php artisan aimeos:setup --option=setup/default/demo:1
> php artisan aimeos:account --super <email address>
> ```

Thanks for the quick response! When I got your answer, I've already tried another way to install aimeos, but failed also. So, I had no choice to try the steps above.
I've deleted all the installation files, and run `composer create-project aimeos/aimeos myshop` again. The installation process had no errors any more, the site is working now.
Just one thing I forgot to mention and can make sense for you: the error during my first try of installation has come when I hit the return key after entering the value for DB_USERNAME. The install script should have stepped to asking for DB_PASSWORD instead of giving that error. But now it works fine for me.
